tutorials/firefly/OpenUniverse2024Preview_Firefly.md

```diff
@@ -0,0 +1,750 @@
+
+- Learn how to launch an interactive Firefly instance inside JupyterLab.
+
+- Learn how to use the Firefly Jupyterlab extension to visualize cloud-hosted simulated images, overplot ds9 regions, overplot catalogs in Parquet format, and create 3 color images.
+
++++
+
+## Introduction
+
+The purpose of this tutorial is to become familiar with simulated Roman and Rubin simulated data published through the OpenUniverse 2024 data preview, and to become familiar with the Firefly JupyterLab Extension for visualizing astronomical data products.
+
+OpenUniverse2024 is a project to simulate spatially overlapping imaging surveys to be carried out by the Nancy Grace Roman Telescope and the Vera C. Rubin Observatory. The simulations were carried out on Argonne's Theta cluster and consist of:
+
+- The LSST ELAIS-S1 Deep Drilling Field (DDF)
+- The Roman Time-Domain Survey (TDS) shifted to overlap the ELAIS region and LSST DDF
+- Overlapping LSST Wide-Fast-Deep (WFD) survey (with rolling cadence)
+- Overlapping Roman Wide-Area Survey (WAS) in the same region
+- A deep-field calibration region of the Roman WAS in the same region
+
+This data preview release consists of a subset of data from each of the five categories above. More information about data preview can be found at:
+https://irsa.ipac.caltech.edu/data/theory/openuniverse2024/overview.html
+
+Firefly is an open-source web-based UI library for astronomical data archive access and visualization developed at Caltech and used by multiple space- and ground-based astrophysics archives. More information can be found here:
+https://github.com/Caltech-IPAC/firefly/blob/dev/README.md
+
+In addition to being used to make web applications, Firefly can be used from Python. More information can be found here:
+https://caltech-ipac.github.io/firefly_client/usage/index.html
+
+The Firefly JupyterLab Extension makes it particularly easy to use Firefly to efficiently visualize cloud-hosted astronomical data using cloud-based JupyterLab instances. More information can be found here:
+
+https://github.com/Caltech-IPAC/jupyter_firefly_extensions/blob/master/README.md
+
++++
+
+## Imports
+
+- astropy.io.fits for accessing FITS files
+- numpy for numerical computing
+- s3fs for browsing cloud buckets
+- matplotlib.pyplot for creating static visualizations of FITS images
+- matplotlib.patches for annotating visualizations of FITS images
+- astropy.wcs for dealing with astronomical world coordinate systems
+- astropy.units for dealing with astronomical units
+- astropy.coordinates.SkyCoord for dealing with astronomical coordinates
+- firefly_client.FireflyClient for using the Firefly python client
+- astropy.nddata.Cutout2D for making image cutouts
+- itertools.product to support looping over Roman blocks
+- reproject.reproject_interp to convert Roman coadds from STG to TAN projection
+- io.BytesIO for writing a fits file to an in-memory stream
+
+```{code-cell} ipython3
```

Comment:
A cell with a commented out `pip install ` content is missing. Please add that in the same style as the other notebooks have (commented out list of all the directly imported dependencies. If firefly require something else that is not directly imported but is not installed as a dependency of something, then list those, too)

Reply:
Oh, ok - I'll add it